src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_AND_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
replace this checking `if (storageDao.get(user.getLogin()) != null)` after `if (user.getLogin() == null)` because you can get an exception when you call user.getLogin() and it could be null